File: Array & Vector & Hash/1. TwoSum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        search_map = {}
        ans = []

        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in search_map:
                ans.append(search_map[complement])
                ans.append(i)
                return ans
            else:
                search_map[nums[i]] = i

        return ans

File: Array & Vector & Hash/1. TwoSum/1. Two Sum.cpp
class Solution {
    public:
        vector<int> twoSum(vector<int>& nums, int target) {
            /* declare map and answer vector */
            unordered_map<int, int> search_map;
            vector<int> ans;
            int size = nums.size();
            /* for loop nums */
            for (int index = 0; index < size; index++) {
                int complement = target - nums[index];
                /* 
                    map[key] = value
                    find() search for "key"
                */
                if (search_map.find(complement) != search_map.end()) {
                    /* insert complement's index */
                    ans.push_back(search_map[complement]);
                    /* insert current index */
                    ans.push_back(index);
                    return ans;
                } else {
                    /* if not found，insert current num to key */
                    search_map[nums[index]] = index;
                }               
            }
            return ans;
        }
    };